core/src/main/java/org/apache/brooklyn/location/byon/FixedListMachineProvisioningLocation.java

```diff
@@ -279,7 +280,7 @@ public T obtain() throws NoMachinesAvailableException {
     public T obtain(Map<?,?> flags) throws NoMachinesAvailableException {
         T machine;
         T desiredMachine = (T) flags.get("desiredMachine");
-        ConfigBag allflags = ConfigBag.newInstanceExtending(config().getBag()).putAll(flags);
+        ConfigBag allflags = ConfigBag.newInstanceExtending(config().getBag(), flags);
```

Comment:
This change looks like it makes no difference (now that you've backed out your changes to the `newInstanceExtending()` method). I'm fine with leaving this change in though.